pkg/mattermost/mattermost.go

```diff
@@ -0,0 +1,236 @@
+		return
+	}
+
+	// Check if Channel exists or create Channel and add user in Mattermost
+	botChannel, err = getBotChannel(b.ChannelName, botTeam.Id, botUser.Id)
+	if err != nil {
+		logging.Logger.Error("There was a problem creating channel. Error: ", err)
+		return
+	}
+
+	// Create WebSocketClient and handle messages
+	webSocketURL := WebSocketProtocol + strings.SplitN(b.ServerURL, ":", 2)[1]
+	webSocketClient, _ := model.NewWebSocketClient4(webSocketURL, client.AuthToken)
+	webSocketClient.Listen()
+	go func() {
+		for {
+			select {
+			case event := <-webSocketClient.EventChannel:
+				handleMMessage(event, b.AllowKubectl, b.ClusterName, b.ChannelName)
+			}
+		}
+	}()
+	return
+}
+
+// Check if Mattermost server is reachable
+func checkServerConnection() error {
+	if _, resp := client.GetOldClientConfig(""); resp.Error != nil {
+		return resp.Error
+	}
+	return nil
+}
+
+// Check if team exists in Mattermost
+func getBotTeam(teamName string) (*model.Team, error) {
+	botTeam, resp := client.GetTeamByName(teamName, "")
+	if resp.Error != nil {
+		return botTeam, resp.Error
+	}
+	return botTeam, nil
+}
+
+// Create botkube user if not present
+func createBotUser(teamID, email string) (*model.User, error) {
+	// Check Botkube user in system
+	botUser, resp := client.GetUserByUsername(BotName, "")
+	if resp.Error != nil {
+		botUser, resp = client.CreateUser(mmBot(email))
+		if resp.Error != nil {
+			return nil, resp.Error
+		}
+	}
+	if botUser.Email != email {
+		logging.Logger.Warn("Email address of user doesn't match with the provided Email")
+	}
+	// Check Botkube user in team
+	users, _ := client.SearchUsers(searchBot(teamID, BotName))
+	if len(users) > 0 {
+		logging.Logger.Info("User already exists in Mattermost team")
+		return botUser, nil
+	}
+	// Add Botkube user
+	if _, resp = client.AddTeamMember(teamID, botUser.Id); resp.Error != nil {
+		return nil, resp.Error
+	}
+	logging.Logger.Info("User added to Mattermost successfully")
+	return botUser, nil
+}
+
+// Create bot channel if not present and add botkube user in channel
+func getBotChannel(channelName, botTeamID, botUserID string) (*model.Channel, error) {
+	// Checking if channel exists
+	botChannel, resp := client.GetChannelByName(channelName, botTeamID, "")
+	if resp.Error != nil {
+		logging.Logger.Debug("Channel not present in Mattermost. Creating channel")
+		// Creating channel
+		if botChannel, resp = client.CreateChannel(mmChannel(channelName, botTeamID)); resp.Error != nil {
+			return botChannel, resp.Error
+		}
+	}
+	logging.Logger.Info("Channel created successfully in Mattermost")
+
+	// Adding Botkube user to channel
+	_, resp = client.AddChannelMember(botChannel.Id, botUserID)
+	logging.Logger.Info("Botkube User added to the Channel")
+	return botChannel, nil
+}
+
+// Check incomming message and take action
+func handleMMessage(event *model.WebSocketEvent, allowkubectl bool, clusterName, channelName string) {
```

Comment:
can be handleMessage only as its in mattermost package and private function. Same for rest of private functions

Reply:
Yes. All the functions in mattermost package will be private.